sdk/tests/conformance2/uniforms/large-uniform-buffers.html

```diff
@@ -0,0 +1,155 @@
+**
+** THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+** EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+** MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+** IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+** CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+** TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+** MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+*/
+
+-->
+
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<title>WebGL large uniform buffers conformance tests</title>
+<link rel="stylesheet" href="../../resources/js-test-style.css"/>
+<script src="../../js/js-test-pre.js"></script>
+<script src="../../js/webgl-test-utils.js"></script>
+<script id="vshader" type="x-shader/x-vertex">#version 300 es
+in vec4 a_vertex;
+void main(void) {
+  gl_Position = a_vertex;
+}
+</script>
+<script id="fshader" type="x-shader/x-fragment">#version 300 es
+precision mediump float;
+layout (std140) uniform color_ubo {
+  vec4 color;
+};
+out vec4 fragColor;
+void main(void) {
+  fragColor = color;
+}
+</script>
+</head>
+<body>
+<canvas id="example" width="100", height="100"></canvas>
+<div id="description"></div>
+<div id="console"></div>
+<script>
+"use strict";
+debug("");
+description("Test for UBOs sized over 65536 - https://bugs.chromium.org/p/angleproject/issues/detail?id=3388");
+
+var wtu = WebGLTestUtils;
+var gl = wtu.create3DContext("example", undefined, 2);
+
+function runTest() {
+    var program = wtu.setupProgram(gl, ['vshader', 'fshader'], ['a_vertex']);
+    var uboIndex = gl.INVALID_INDEX;
+    if (program)
+        uboIndex = gl.getUniformBlockIndex(program, "color_ubo");
+    if (!program || uboIndex == gl.INVALID_INDEX) {
+        testFailed("Loading program failed");
+        return;
+    }
+    testPassed("Loading program succeeded");
+
+    var vertices = new Float32Array([
+        -1, -1, 0,
+         1, -1, 0,
+        -1,  1, 0,
+         1,  1, 0
+    ]);
+    var vertexBuf = gl.createBuffer();
+    gl.bindBuffer(gl.ARRAY_BUFFER, vertexBuf);
+    gl.bufferData(gl.ARRAY_BUFFER, vertices, gl.STATIC_DRAW);
+    gl.enableVertexAttribArray(0);
+    gl.vertexAttribPointer(0, 3, gl.FLOAT, false, 0, 0);
+
+    var indexBuf = gl.createBuffer();
+    gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, indexBuf);
+    gl.bufferData(gl.ELEMENT_ARRAY_BUFFER, new Int16Array([ 0, 1, 2, 2, 1, 3 ]), gl.STATIC_DRAW);
+
+    var uboDataSize = gl.getActiveUniformBlockParameter(
+        program, uboIndex, gl.UNIFORM_BLOCK_DATA_SIZE);
+    if (uboDataSize == 0) {
+        testFailed("uniform block data size invalid");
+        return;
+    }
+
+    var uboBuf = gl.createBuffer();
+    var uboData = new Float32Array(0x20000);
+
+    var offs0 = 0x00000;
+    // Red
+    uboData[offs0 + 0] = 1;
+    uboData[offs0 + 1] = 0;
+    uboData[offs0 + 2] = 0;
+    uboData[offs0 + 3] = 1;
+
+    var offs1 = 0x10000;
```

Comment:
offs1 has to be computed as a multiple of gl.getParameter(UNIFORM_BUFFER_OFFSET_ALIGNMENT), or the test might be incorrect on some platforms.

Reply:
The maximum alignment ever seen in the wild is 256 (by NVIDIA GPUs): http://opengl.gpuinfo.org/displaycapability.php?name=GL_UNIFORM_BUFFER_OFFSET_ALIGNMENT , and there is also GLES spec language capping it to 256. The `65536` size isn't an OpenGL limit, it's a Direct11 cbuffer limit, so there's no easy way I can compute this offset from the details given to me from GLES alone.

Reply:
At least could you please add an assertion that offs1 is evenly divisible by UNIFORM_BUFFER_OFFSET_ALIGNMENT? Thanks.

Reply:
Sounds good. Any recommended ways to test an assertion? I'll tackle this tomorrow.

Reply:
const alignment = gl.getParameter(gl.UNIFORM_BUFFER_OFFSET_ALIGNMENT);
if (offs1 % alignment != 0) {
  testFailed("Platform has a strange uniform buffer offset alignment: " + alignment);

or similar.